File: Python3/프로그래머스/3/64062. 징검다리 건너기/징검다리 건너기.py
def check(mid, stones, k):
    count = 0
    for stone in stones:
        if stone < mid:
            count += 1
        else:
            count = 0
        if count == k:
            return False
    return True

def solution(stones, k):
    left, right = 1, max(stones)
    
    while left <= right:
        mid = (left + right) // 2
        if check(mid, stones, k):
            left = mid + 1
        else:
            right = mid - 1
            
    return right

File: Java/프로그래머스/3/64062. 징검다리 건너기/징검다리 건너기.java
class Solution {	
    public int solution(int[] stones, int k) {
    	
    	int l = 1;
    	int r = Integer.MIN_VALUE;
    	
    	for (int i = 0; i < stones.length; i++) {
    		if (r < stones[i]) {
    			r = stones[i];
    		}
		}
    	
    	while (l <= r) {
    		int mid = (l + r) / 2;
    		
    		int count = 0;
    		boolean check = false;
    		
    		for (int i = 0; i < stones.length; i++) {
				if (stones[i] < mid) {
					count++;
				} else {
					count = 0;
				}
				
				if (count == k) {
					r = mid - 1;
					check = true;
					break;
				}
			}
    		
    		if (!check) {
    			l = mid + 1;
    		}
    	}
    	
    	return r;
    }
}